src/System.Management.Automation/engine/Modules/ModuleUtils.cs

```diff
@@ -66,9 +69,11 @@ internal static bool IsPossibleModuleDirectory(string dir)
         /// Get all module files by searching the given directory recursively.
         /// All sub-directories that could be a module folder will be searched.
         /// </summary>
-        internal static IEnumerable<string> GetAllAvailableModuleFiles(string topDirectoryToCheck)
+        internal static IEnumerable<string> GetAllAvailableModuleFiles(string topDirectoryToCheck, bool skipEditionCheck)
```

Comment:
Let's defer solving the `Get-Module -List -All` problem. We are not sure what is the right user experience yet. We can solve it as a bug fix.

Reply:
Please open tracking and reference issue.

Reply:
I think we agreed to not solve `Get-Module -List -All` within this PR, because we are not sure what's the right UX.
Anything changes since we last discussion?

Reply:
Please see https://github.com/PowerShell/PowerShell/issues/7297